### DISCUSSION TITLE
Option to Automatically Use Split Frame and Locked Group in VS Code

### DISCUSSION THREAD

**Discussion by jerrens** [bendera/vscode-commit-message-editor#61] [Ideas]
Hi,

First of all, thank you for this plugin!

I was wondering if the APIs were available for the commit message tab created by this plugin to utilize the split editor feature and also the more recently added Lock Group option.  

When writing check-in comments, I like to step through the staged files and review the changes in the diff, and add related comments to explain the changes.  Instead of switching back and forth between tabs in the editor, I normally arrange my tabs like the following:

![image](https://user-images.githubusercontent.com/11966839/178000814-8a6f702c-9f1a-4b96-a740-a49701bf31eb.png)

By using the 'Lock Group' option for the group (of one tab) where the commit message tab is, when selecting the staged files in the source control window, their diff appears in the top panel of the editor instead of the active group (which is normally the commit message since the textarea has focus from where I was entering comments).

It would be a helpful feature (at least for me) if this were an option that could be enabled for the Commit Message Editor - to automatically open the commit message editor in a split pane that was locked from other tabs being automatically added.

**Comment by bendera** [comment]
Hi,

I'm glad to like my extension.

First of all, I need to know this "Lock Group" feature, I've never used it. I've found an "Auto Lock Groups" section in the settings page but I don't know if I could add the message tab to that list. Let me know if you found an extension that is able to do that.

**Reply by jerrens** [reply]
Here is a link to the release notes where the feature came out of preview:
https://code.visualstudio.com/updates/v1_61#_locked-editor-groups

From using it, I assume it is a property that can be set to any editor group.  I haven't gotten into writing VS code extensions (yet), so I'm not sure where to dig in to find their APIs.  If when opening a new tab in the editor, you can specify that it be opened in a new tab group, and get reference to the tab or group that was created, hopefully the property is somewhere in that instance you get reference to. 

Thanks for investigating!

**Comment by bendera** [comment]
This is what I found so far:

There is an option to control in which column to place the tab.
https://code.visualstudio.com/api/references/vscode-api#ViewColumn
In your case, the bottom panel is the "Three". I can add a config option to set this.

In theory a webview panel can be locked automatically but I couldn't do it.
https://github.com/microsoft/vscode/issues/125791

**Comment by bendera** [comment]
UPDATE

Instead of opening in a tab group, it could also open in the bottom panel area (next to the "PROBLEMS", "OUTPUT", etc.). I will implement this feature.

**Reply by jerrens** [reply]
I'll be interested to see this solution.  It sounds like it would work well to provide what I was looking for.

Thanks

**Reply by jerrens** [reply]
Was the version that included "Add ability to open the editor in a specific editor group."  referencing this feature?

I saw the setting for `commit-message-editor.view.columnToShowIn` appear in that release (v0.23.0), but the options available (Active, Beside, One...Nine) only seem to only add a new tab in the editor (legacy method) and all of the other options trigger a split-screen.  I can't seem to find the option to have it open down in the "Panel" area.  Maybe this still a work in progress, or maybe I'm missing something?

**Reply by jerrens** [reply]
I ran across [this post on StackOverflow](https://stackoverflow.com/questions/75808371/how-can-i-run-multiple-commands-with-a-single-vs-code-keybinding-without-an-exte) and decided to create a keyboard shortcut (`keybindings.json`) that I can use for now:

```
{
    "key": "ctrl+shift+alt+c",
    "command": "runCommands",
    "args": {
      "commands":[
        "workbench.action.focusLastEditorGroup", // Move to the last group
        "workbench.action.newGroupBelow", // Create a new group
        "commitMessageEditor.openEditor", // Open Commit Message Editor (In active group)
        "workbench.action.lockEditorGroup", // Now lock it
        "workbench.action.closePanel" // Make some more room
      ]
    }
  }
```

**Reply by bendera** [reply]
> Was the version that included "Add ability to open the editor in a specific editor group." referencing this feature?

Yes. It's only a quick workaround because it was easy to implement. The editor tab will be opened in the editor area but not in the panel view, although it would be the best solution. Unfortunately, I have limited free time to maintain my pet projects, but I want to implement this feature, definitely.

**Reply by jerrens** [reply]
> Unfortunately, I have limited free time to maintain my pet projects, but I want to implement this feature, definitely.

I completely understand.  That keyboard shortcut is working well for me.  The `runCommands` seems to be the trick I needed.  The [SO answer](https://stackoverflow.com/a/75808372/2136313) also provides an example for extension developers.  I haven't explored extension development yet, so I'm not sure if that is any help for you or not.  The indirect method of stacking those commands seems like it may be fragile depending on how others may have their editors arranged.

Thanks for what you have done on the extension so far!
